tests/test_resources/test_modules/test_functions/test_aws_lambda/test_aws_lambda.py

```diff
@@ -0,0 +1,554 @@
+    For example, there are a.py, b.py and c.py. Each file has the following funcs, respectively: func_a, func_b and
+    func_c. So in the test, the main function (handler) will look something like this:
+    import c.py
+    def handler_func:
+        func_c()
+    where func_c() calls func_b() which calls func_a().
+    """
+    prefix = "call_files_chain"
+    folder_path = f"{TEST_RESOURCES}/{prefix}"
+    handler_paths = os.listdir(folder_path)
+    handler_paths = [p for p in handler_paths if ".py" in p]
+    handler_paths.sort()
+    handler_paths = [f"{folder_path}/{p}" for p in handler_paths]
+    name = "test_lambda_multiple_files_c"
+    my_lambda_calc_2 = rh.aws_lambda_function(
+        paths_to_code=handler_paths,
+        handler_function_name="special_calc",
+        runtime="python3.9",
+        args_names=["arg1", "arg2"],
+        name=name,
+    )
+    res1 = my_lambda_calc_2(2, 3)
+    res2 = my_lambda_calc_2(5, 3)
+    res3 = my_lambda_calc_2(2, 7)
+    res4 = my_lambda_calc_2(10, 5)
+    assert res1 == "16"
+    assert res2 == "17"
+    assert res3 == "20"
+    assert res4 == "20"
+    LAMBDAS_NAMES.add(my_lambda_calc_2.name)
+
+
+def test_args():
+    basic_func = rh.aws_lambda_function(name="test_lambda_create_and_run")
+    res1 = basic_func(2, 3)
+    res2 = basic_func(5, arg2=3)
+    res3 = basic_func(arg1=2, arg2=7)
+    assert res1 == "5"
+    assert res2 == "8"
+    assert res3 == "9"
+
+
+def test_map_starmap():
+    basic_func = rh.aws_lambda_function(name="test_lambda_create_and_run")
+    res_map1 = basic_func.map([1, 2, 3], [4, 5, 6])
+    res_map2 = basic_func.map([6, 2, 3], [15, 52, 61])
+    res_map3 = basic_func.starmap([(1, 2), (3, 4), (5, 6)])
+    res_map4 = basic_func.starmap([(12, 5), (44, 32), (8, 3)])
+    assert res_map1 == ["5", "7", "9"]
+    assert res_map2 == ["21", "54", "64"]
+    assert res_map3 == ["3", "7", "11"]
+    assert res_map4 == ["17", "76", "11"]
+
+
+def test_create_from_config():
+    handler_path = [f"{TEST_RESOURCES}/basic_test_handler.py"]
+    name = "test_lambda_from_config"
+    config = {
+        "paths_to_code": handler_path,
+        "handler_function_name": "lambda_sum",
+        "runtime": "python3.9",
+        "args_names": ["arg1", "arg2"],
+        "name": name,
+        "tmp_size": 3072,
+    }
+    config_lambda = rh.AWSLambdaFunction.from_config(config)
+    res1 = config_lambda(1, 2)
+    res2 = config_lambda(8, 12)
+    res3 = config_lambda(14, 17)
+
+    assert res1 == "3"
+    assert res2 == "20"
+    assert res3 == "31"
+    LAMBDAS_NAMES.add(config_lambda.name)
+
+
+def test_share_lambda():
+    basic_func = rh.aws_lambda_function(name="test_lambda_create_and_run")
+    users = ["josh@run.house"]
+    added_users, new_users = basic_func.share(
+        users=users, notify_users=True, access_type="write"
+    )
+    assert added_users == {}
+    assert new_users == {}
+
+
+def test_remove_resources():
```

Comment:
do we have functionality / testing around deleting the function? (similar to what we do with clusters with `teardown_and_delete`, where we delete the resources themselves from lambdas along with the config from RNS